raven-android/src/main/java/com/getsentry/raven/android/event/helper/AndroidEventBuilderHelper.java

```diff
@@ -291,29 +290,51 @@ private static Boolean isCharging(Context ctx) {
      * @return the device's current kernel version, as a string (from uname -a)
      */
     private static String getKernelVersion() {
-        String errorMsg = "Exception while attempting to read kernel information";
-        BufferedReader br = null;
         try {
-            Process p = Runtime.getRuntime().exec("uname -a");
-            if (p.waitFor() == 0) {
-                br = new BufferedReader(new InputStreamReader(p.getInputStream()));
-                return br.readLine();
-            }
+            return formatKernelVersion(Util.readLine("/proc/version"));
         } catch (Exception e) {
-            Log.e(TAG, errorMsg, e);
-        } finally {
-            if (br != null) {
-                try {
-                    br.close();
-                } catch (IOException ioe) {
-                    Log.e(TAG, errorMsg, ioe);
-                }
-            }
+            Log.e(TAG, "Exception while attempting to read kernel information", e);
         }
 
         return null;
     }
 
+    private static String formatKernelVersion(String rawKernelVersion) {
+        // Example (see tests for more):
+        // Linux version 3.0.31-g6fb96c9 (android-build@xxx.xxx.xxx.xxx.com) \
+        //     (gcc version 4.6.x-xxx 20120106 (prerelease) (GCC) ) #1 SMP PREEMPT \
+        //     Thu Jun 28 11:02:39 PDT 2012
+
+        final String procVersionRegex =
+                "Linux version (\\S+) " + /* group 1: "3.0.31-g6fb96c9" */
+                "\\((\\S+?)\\) " +        /* group 2: "x@y.com" (kernel builder) */
+                "(?:\\(gcc.+? \\)) " +    /* ignore: GCC version information */
+                "(#\\d+) " +              /* group 3: "#1" */
+                "(?:.*?)?" +              /* ignore: optional SMP, PREEMPT, and any CONFIG_FLAGS */
+                "((Sun|Mon|Tue|Wed|Thu|Fri|Sat).+)"; /* group 4: "Thu Jun 28 11:02:39 PDT 2012" */
+
+        // CHECKSTYLE.OFF: MagicNumber
+        final int procVersionGroupKernelVersion = 1;
+        final int procVersionGroupBuilderName = 2;
+        final int procVersionGroupBuildNumber = 3;
+        final int procVersionGroupBuildDate = 4;
+        final int procVersionGroupCount = 4;
+        // CHECKSTYLE.ON: MagicNumber
+
+        Matcher m = Pattern.compile(procVersionRegex).matcher(rawKernelVersion);
+        if (!m.matches()) {
+            Log.e(TAG, "Regex did not match on /proc/version: " + rawKernelVersion);
+            return "Unavailable";
```

Comment:
I think we should return `null` on both of these cases, because `null` is the common "couldn't retrieve" value in the Sentry UI.